src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,126 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
 public class StringManipulation implements StringManipulationInterface {
+    private String message;
 
     @Override
     public String getString() {
-        return null;
+        return message;
     }
 
     @Override
     public void setString(String string) {
+        message = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        int counting = 0;
+        for (int i = 0; i < message.length(); i++) {
+            if (message.charAt(i) == ' ') {
+                if (counting == 0) {
+                    if (i > 0 && message.charAt(i - 1) != ' ') {
+                        counting = 1;
+                    }
+                }
+                if (i + 1 < message.length() && message.charAt(i + 1) != ' ') {
+                    counting++;
+                }
+            }
+        }
+        return counting;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+
+        if (n > message.length()) {
+            throw new IndexOutOfBoundsException("N is greater than string length!");
+        }
+        if (n <= 0) {
+            throw new IllegalArgumentException("N is less than or equal to 0");
+        }
+
+        List<String> array = new ArrayList<>(message.length() + n - 1);
+
+        for (int start = 0; start < message.length(); start += n) {
+            array.add(message.substring(start, Math.min(message.length(), start + n)));
+        }
+
+        for (int i = 0; i < array.size(); i++) {
+            if (array.get(i).length() == n) {
+                String temp = array.get(i);
+                temp = temp.substring(0, n - 1);
+                if (maintainSpacing) {
+                    temp = temp + " ";
+                }
+                array.set(i, temp);
+            }
+        }
+        String a = "";
+        for (int i = 0; i < array.size(); i++) {
+            a = a + array.get(i);
+        }
+        return a;
     }
 
+
     @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+
+        if (startWord <= 0 ||endWord <= 0 || startWord >endWord) {
+            throw new IllegalArgumentException("Invalid endWord/startWord");
+        }
+
+        if (count() < endWord) {
+            throw new IndexOutOfBoundsException("String has less words than endWords amount in it!");
+        }
+
+        String after = message.trim().replaceAll(" +", " ");
+
+        List<String> wordArrayList = new ArrayList<String>();
+        for (String word : after.split(" ")) {
+            wordArrayList.add(word);
+        }
+        int test = wordArrayList.size();
+        String[] aa = new String[test];
```

Comment:
Instead of converting the ArrayList to an array using a loop, you can directly use the toArray method provided by the ArrayList class.